raster/r.surf.gauss/main.c

```diff
@@ -55,9 +58,43 @@ int main(int argc, char *argv[])
     sigma->type = TYPE_DOUBLE;
     sigma->answer = "1.0";
 
+    seed = G_define_option();
+    seed->key = "seed";
+    seed->type = TYPE_INTEGER;
+    seed->required = NO;
+    seed->label = _("Seed for random number generator");
+    seed->description = _("The same seed can be used to obtain same results"
+                          " or random seed can be generated by other means.");
+
+    s_flag = G_define_flag();
+    s_flag->key = 's';
+    s_flag->label = _("Generate random seed");
+    s_flag->description =
+        _("Automatically generates random seed for random number"
+          " generator (use when you don't want to provide the seed option)");
+
     if (G_parser(argc, argv))
         exit(EXIT_FAILURE);
 
+    /****** INITIALISE RANDOM NUMBER GENERATOR ******/
+    if (s_flag->answer) {
+        seed_value = G_math_srand_auto();
+        G_verbose_message(_("Generated random seed (-s): %ld"), seed_value);
+    }
+    else if (seed->answer) {
+        seed_value = atol(seed->answer);
+        G_math_srand(seed_value);
+        G_verbose_message(_("Read random seed from %s option: %ld"), seed->key,
+                          seed_value);
+    }
+    else {
+        /* default as it used to be */
+        seed_value = G_math_srand_auto();
+        G_verbose_message(_("Warning set flag s or option seed. Generated "
+                            "random seed (-s): %ld"),
+                          seed_value);
```

Comment:
This should probably be G_warning, no?

Reply:
Speaking from from strict API point of view, it should be message, later warning, and when possible to break compatibly an error. I would have to investigate to see what we did in the past (if message or warning).